src/Ontology/Account.h

```diff
@@ -0,0 +1,47 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include <array>
+#include <vector>
+#include <string>
+
+#include "Address.h"
+#include "PrivateKey.h"
+#include "BinaryCoding.h"
+
+namespace TW {
+namespace Ontology {
+
+class Account {
+private:
+
+    Data publicKey;
+    Data privateKey;
+    std::string address;
+
+public:
+
+    explicit Account(const std::string &priKey);
+
+    explicit Account(const Data &priKey);
+
+    PrivateKey getPrivateKey() const;
+
+    PublicKey getPublicKey() const;
+
+    Address getAddress() const;
+
+    Data sign(const Data &msg) noexcept;
```

Comment:
maybe a static `sign` method is enough? like this `Signer.sign(Transaction, PrivateKey)`

Reply:
I think this maybe not enough. In ONT & ONG transaction, we do the following thing in `Accout `class: 

1. using `Account `class to generate `Address` which based on public key, the `Address` will be used to generate contract invoke script (both ONT & ONG and based on go smart contract).
2. using `Account `class to generate signature for transaction. In general, a transaction may need two  `Account ` to sign, one to prove asset operation is legal, another to pay for the gas fee.

You can click [here](https://github.com/NashMiao/wallet-core/blob/master/src/Ontology/Ont.cpp) to review my `ONT `class, and click [here](https://github.com/NashMiao/wallet-core/blob/master/src/Ontology/Ong.cpp) to review my `ONG `class.

The signature information is a part of transaction, which defined in `Transaction` class in privately. Because of this reason, we defined two interface for `Accout `to add signature in `Transaction`:
 
- void sign(const Account &acct);
- void addSign(const Account &acct);

Reply:
Ok, How about `Signer.sign(transaction, privateKey, payerPrivateKey)`? Transaction is only used for serialization, basically you can get public key and address from a private key

Reply:
In a transaction, up to 16 signature we support. How about I defined a `Signer` class, and provide two interface for add signature:

- `Signer.sign(transaction, privateKey)`
- `Signer.addSign(transaction, privateKey)`

Reply:
Here is the `Signer` class with `sign`, `addSign` method. https://github.com/TrustWallet/wallet-core/pull/201/commits/da6edb41083bb71a8cf243933bc85c49870c0255